android/src/main/java/com/lykhonis/flutterbilling/BillingPlugin.java

```diff
@@ -190,8 +205,12 @@ public void failed() {
 
         final List<String> identifiers = new ArrayList<>(purchases.size());
 
+        Log.d(TAG, "fetchPurchases, trying to consume if necessary:");
+
         for (Purchase purchase : purchases) {
             identifiers.add(purchase.getSku());
+            if (consumeIds.indexOf(purchase.getSku()) != -1)
```

Comment:
Should this be a set and `if (consumeIds.remove(purchase.getSku())` then consume?

Reply:
I see what you mean, but in fact there is no reason for removing objects into consumeIds, right? For a given id/sku, user want it to be always consume or never ; it won't change.

Reply:
I guess I assumed that since we have a `consume` parameter now, we want it to be one-shot so we do not retain state on this level. If there is new purchase request, that request must come with `consume = true` in order to consume it. But in general, yes, I see what you are saying. On other note, I would still suggest using Set and use `contains`/`remove` instead of `indexOf`